test/jdk/jdk/incubator/vector/AbstractVectorConversionTest.java

```diff
@@ -335,10 +368,23 @@ static void zeroArray(Object a, int offset, int length) {
     static void copyConversionArray(Object src, int srcPos,
                                     Object dest, int destPos,
                                     int length,
+                                    VectorSpecies srcSpecies,
+                                    VectorSpecies dstSpecies,
                                     Function<Number, Object> c) {
+        if (srcSpecies.elementType() == dstSpecies.elementType()) {
+            System.arraycopy(src, srcPos, dest, destPos, length);
+            return;
+        }
         for (int i = 0; i < length; i++) {
             Number v = (Number) Array.get(src, srcPos + i);
-            Array.set(dest, destPos + i, c.apply(v));
+            if (srcSpecies.elementType() == Halffloat.class) {
+                 v = (Number) Halffloat.valueOf(v.shortValue()).floatValue();
```

Comment:
Any specific reason for not directly calling Float.float16ToFloat(short) here instead of creating a Halffloat instance ?

Reply:
Done